### PR TITLE
Normalise input newlines

### LINKED ISSUE #797: Feature request: Enable use of unix-style line endings when running sqlfluff fix command on Windows
In our environment, we use Windows PCs for development, but the company standard is that SQL files have unix-style line endings (LF).  We use bash under WSL a lot, and set Windows editors to use LF only.  However, the only python we have available is on Windows. When running sqlfluff fix under python on Windows, the file gets changed from LF to Windows-style line endings CRLF.

I understand that this is python default behavior, but it would be nice to be able to over-ride the default and choose LF only, all the time.

### PR DESCRIPTION
**Pull request by jpy-git**
### Brief summary of the change made

Fixes #797. After investigation (described in the issue) it seems most appropriate to not handle multiple newlines and instead add step to ensure all newline inputs to sqlfluff are normalised to unix-style line endings for consistency. Users wanting newline conversion should consider the easy to use [mixed-line-ending](https://github.com/pre-commit/pre-commit-hooks/blob/master/pre_commit_hooks/mixed_line_ending.py) pre-commit hook or the `dos2unix` command. 

Currently we load from files using universal newlines, but we don't do this for test input. Therefore I have added a conversion function to the linter to ensure consistency regardless of how the SQL is supplied.

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `python test/generate_parse_fixture_yml.py` or by running `tox` locally).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by tunetheweb** [comment]
Isn’t there a concern here that `sqlfluff fix` will now effectively rewrite line endings and lead to massive diffs every time a file needs a fix? At least until the code base is converted to unix style newlines in entirety?

Note I’m not talking about our code base (that I already converted) but any other users.

**Comment by jpy-git** [comment]
@tunetheweb python already automatically does this when it reads in files so it's no different from what we already do, I'm just making it consistent for the API and CLI stdin.

**Comment by jpy-git** [comment]
also when python writes out a file it automatically converts them back to the appropriate line ending for the system

**Comment by jpy-git** [comment]
@barrywhart follow up PR to my post in #797

**Comment by tunetheweb** [comment]
> @tunetheweb python already automatically does this when it reads in files so it's no different from what we already do, I'm just making it consistent for the API and CLI stdin.

I'm confused. If python already automatically does this, then why's there even an issue? Sorry if I'm missing the obvious here...

**Comment by jpy-git** [comment]
@tunetheweb because it only does it when reading in from a file (i.e. `open("<some_sql_path>", "r") as f`), but not when we read the sql from either stdin in the CLI or via the string input to the API. So we should normalise those too for consistency

**Comment by tunetheweb** [comment]
OK. Thanks for explaining.

File: src/sqlfluff/core/linter/linter.py [hunk 1]
```diff
@@ -116,6 +116,11 @@ def _load_raw_file_and_config(
         # Return the raw file and config
         return raw_file, file_config, encoding
 
+    @staticmethod
+    def _normalise_newlines(string: str) -> str:
+        """Normalise newlines to unix-style line endings."""
+        return regex.sub(r"\n|\r\n|\r", "\n", string)
```
Comment (tunetheweb) [inline src/sqlfluff/core/linter/linter.py:122]: Do we need this first one? Seems to be replacing `"\n"` with `"\n"`?

```suggestion
        return regex.sub(r"\r\n|\r", "\n", string)
```

**Review by tunetheweb — APPROVED** [review]
One small question but other than that this looks good.

File: src/sqlfluff/core/linter/linter.py (hunk 1 above)
Comment (jpy-git) [reply src/sqlfluff/core/linter/linter.py:122]: strictly no, but thought it was clearer to read as replacing all newlines with the unix newline (i.e. normalise), although in this case it's not like we'll ever change the newline replace style so happy either way.

I'll let you commit the suggestion and merge if you want

**Pull request merged**